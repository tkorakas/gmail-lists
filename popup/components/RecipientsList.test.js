import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import RecipientsList from './RecepientsList';
import chrome from '../../utils/ChromeMock';

describe('RecipientsList component', () => {
  beforeEach(() => {
   window.chrome = chrome;
  });


  test('insert new value to list', () => {
    const list = mount(
      <RecipientsList changePage={jest.fn()} item='list' />
    );

    // Find input field.
    const input = list.find('input');
    // Type on field and enter value.
    input.getDOMNode().value = 'email@example.com';
    input.simulate('change', input);
    input.simulate('keypress', {key: 'Enter'});

    // There is one item on the list.
    expect(list.state().items.length).toBe(1);
    expect(list.state().items[0]).toBe('email@example.com');

    // Input field has no value.
    expect(input.getDOMNode().value).toBe('');

    // Item saved on storage.
    chrome.storage.sync.get((data) => {
      expect(data).toEqual({gmail_lists_list: ["email@example.com"]});
    });
  });

  test('insert empty to list', () => {
    const list = mount(
      <RecipientsList changePage={jest.fn()} item='list' />
    );

    // Find input field.
    const input = list.find('input');
    // Type on field and enter value.
    input.getDOMNode().value = '';
    input.simulate('change', input);
    input.simulate('keypress', {key: 'Enter'});

    // There is one item on the list.
    expect(list.state().items.length).toBe(1);

    // Item saved on storage.
    chrome.storage.sync.get((data) => {
      expect(data).toEqual({gmail_lists_list: ["email@example.com"]});
    });

    input.getDOMNode().value = ' ';
    input.simulate('change', input);
    input.simulate('keypress', {key: 'Enter'});

    // There is one item on the list.
    expect(list.state().items.length).toBe(1);

    // Item saved on storage.
    chrome.storage.sync.get((data) => {
      expect(data).toEqual({gmail_lists_list: ["email@example.com"]});
    });
  });

  test('load lists', () => {
    chrome.storage.sync.get((data) => {
      expect(data).toEqual({gmail_lists_list: ["email@example.com"]})
    });

    const list = mount(
      <RecipientsList changePage={jest.fn()} item='list' />
    );

    expect(list.find('li').length).toBe(1);
  });

  test('delete value from list', () =>{
    let list = mount(
      <RecipientsList changePage={jest.fn()} item='list' />
    );

    // Click delete button.
    const deleteButton = list.find('.delete-button').first();
    deleteButton.simulate('click', {targe: {name: 'email@example.com'}});

    // Item saved on storage.
    chrome.storage.sync.get('gmail_lists_list', (data) => {
      expect(data).toEqual({gmail_lists_list: []});
    });
    expect(list.state().items.length).toBe(0);

    list = mount(
      <RecipientsList changePage={jest.fn()} item='list' />
    );
    expect(list.find('li').length).toBe(0);
  });
});
