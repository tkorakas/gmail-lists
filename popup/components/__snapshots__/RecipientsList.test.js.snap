// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipientsList component snapshot test 1`] = `
<div
  className="app-list"
>
  <header
    className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7"
  >
    <div
      className="MuiToolbar-root-36 MuiToolbar-gutters-37"
    >
      <button
        aria-label="Back"
        className="MuiButtonBase-root-44 MuiIconButton-root-38"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginLeft": -14,
            "marginRight": 4,
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-43"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-47"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-53"
        />
      </button>
      <h2
        className="MuiTypography-root-60 MuiTypography-title-66 MuiTypography-colorInherit-79"
      >
        list
      </h2>
    </div>
  </header>
  <div
    className="MuiInput-root-84 MuiInput-fullWidth-91 MuiInput-underline-88"
    onKeyPress={[Function]}
    style={
      Object {
        "marginTop": 16,
        "paddingLeft": 4,
      }
    }
  >
    <input
      autoFocus={true}
      className="MuiInput-input-92 MuiInput-inputType-95"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Recipient email"
      type="email"
    />
  </div>
  <nav
    className="MuiList-root-97 MuiList-padding-98"
  />
</div>
`;
